cli/flox/src/commands/list.rs

```diff
@@ -174,11 +174,17 @@ impl List {
 
             match p {
                 PackageToList::Catalog(_, p) => {
+                    let path_to_display = if let Some(catalog) = &p.catalog {
+                        format!("{}/{}", catalog, p.attr_path)
+                    } else {
+                        p.attr_path.clone()
+                    };
+
                     writeln!(
                         &mut out,
                         "{id}: {path} ({version}{upgrade_available})",
                         id = p.install_id,
-                        path = p.attr_path,
+                        path = path_to_display,
                         version = p.version,
                     )?;
                 },
```

Comment:
`PackageToList::Catalog` actually holds both the manifest entry and the locked entry, only that we chose to ignore the former.
I think (unless we implemented dynamic catalog inference) the right side of the `flox list` output should just take the original `pkg-path` from the descriptor:

https://github.com/flox/flox/blob/80beda03695d79fd0b7ab0a7b07f3fc1021ee320/cli/flox-rust-sdk/src/models/manifest/typed.rs#L505-L522

I do note that all those fields above are crate private. I might have lost the context about it but it should be fine to make them `pub` as the descriptors are only really accessed reading anyway.